Source/ARTDefault.m

```diff
@@ -6,9 +6,35 @@
 //  Copyright (c) 2015 Ably. All rights reserved.
 //
 
-#import "ARTDefault+Private.h"
-
 #import "Ably.h"
+#import "ARTDefault+Private.h"
+#import <sys/utsname.h>
+
+// NSOperatingSystemVersion has NSInteger as version components for some reason, so mitigate it here.
+static UInt32 conformVersionComponent(const NSInteger component) {
+    if (component < 0) return 0;
+    UInt32 conformedValue = (UInt32)component;
+    return conformedValue;
+}
+
+static NSString *canonizeStringAsUserAgentToken(NSString *_Nonnull inputString) {
+    NSMutableString *canonizedString = [NSMutableString string];
```

Comment:
If `canonizedString` (a pointer) is not intended to be modified then please make it const. i.e.:

```suggestion
    NSMutableString *const canonizedString = [NSMutableString string];
```

Same goes for `allowedSet` and `scanner`.

Reply:
https://github.com/ably/ably-cocoa/pull/1119/commits/890c069f75096e056d668f9d45f0300248ffc843